### PR TITLE
Update inputs, outputs & readme

### PR DESCRIPTION
**Pull request by MrMCake**
# Change

Update 
- inputs
- outputs 
- readme

Pipeline reference
[Insights: ActionGroups](https://github.com/Azure/ResourceModules/actions/workflows/ms.insights.actiongroups.yml)

[Insights: ms.insights.activitylogalerts.yml](https://github.com/Azure/ResourceModules/actions/workflows/ms.insights.activitylogalerts.yml)

[Insights: components](https://github.com/Azure/ResourceModules/actions/workflows/ms.insights.components.yml)

[Insights: diagnosticsettings](https://github.com/Azure/ResourceModules/actions/workflows/ms.insights.diagnosticsettings.yml)

[Insights: metricalerts](https://github.com/Azure/ResourceModules/actions/workflows/ms.insights.metricalerts.yml)

[Insights: privatelinkscopes](https://github.com/Azure/ResourceModules/actions/workflows/ms.insights.privatelinkscopes.yml)

[Insights: Scheduledqueryrules](https://github.com/Azure/ResourceModules/actions/workflows/ms.insights.scheduledqueryrules.yml)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

### CODE REVIEW

File: arm/Microsoft.Insights/components/deploy.bicep [hunk 1]
```diff
@@ -67,8 +67,17 @@ module appInsights_rbac '.bicep/nested_rbac.bicep' = [for (roleAssignment, index
...
+@description('The resource ID of the application insights component')
 output appInsightsResourceId string = appInsights.id
+
+@description('The resource group the application insights component was deployed into')
 output appInsightsResourceGroup string = resourceGroup().name
+
+@description('The instrumentation key of the application insights component')
 output appInsightsKey string = appInsights.properties.InstrumentationKey
+
+@description('The application ID of the application insights component')
```
Comment (eriqua) [inline arm/Microsoft.Insights/components/deploy.bicep:82]: These can be accessed through the `existing` keyword in the resource using them and then through dotted access like `<appInsightsResource>.properties.<whatever>`. I'm leveraging this in the web sites refactoring. I suggest to remove those outputs.

File: arm/Microsoft.Insights/components/readme.md [hunk 2]
```diff
@@ -64,13 +64,13 @@ Tag names and tag values can be provided as needed. A tag can be left without a
...
-| `appInsightsName` | string |
-| `appInsightsResourceGroup` | string |
-| `appInsightsResourceId` | string |
+| Output Name | Type | Description |
+| :-- | :-- | :-- |
+| `appInsightsAppId` | string | The application ID of the application insights component |
+| `appInsightsKey` | string | The instrumentation key of the application insights component |
+| `appInsightsName` | string | The name of the application insights component |
+| `appInsightsResourceGroup` | string | The resource group the application insights component was deployed into |
+| `appInsightsResourceId` | string | The resource ID of the application insights component |
```
Comment (eriqua) [inline arm/Microsoft.Insights/components/readme.md:73]: To be updated accordingly after removing additional outputs

File: arm/Microsoft.Insights/metricAlerts/deploy.bicep [hunk 3]
```diff
@@ -1,5 +1,5 @@
 @description('Required. The name of the Alert.')
```
Comment (eriqua) [inline arm/Microsoft.Insights/metricAlerts/deploy.bicep:1]: Alert -> alert for consistency

File: arm/Microsoft.Insights/privateLinkScopes/.parameters/parameters.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
     "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
     "contentVersion": "1.0.0.0",
     "parameters": {
-        "privateLinkScopeName": {
+        "name": {
```
Comment (eriqua) [inline arm/Microsoft.Insights/privateLinkScopes/.parameters/parameters.json:5]: May I hijack this PR for renaming this resource and better align to our naming? 
`sxx-az-pls-x-001`

File: arm/Microsoft.Insights/privateLinkScopes/deploy.bicep [hunk 5]
```diff
@@ -1,6 +1,6 @@
 @description('Required. Name of the Private Link Scope.')
```
Comment (eriqua) [inline arm/Microsoft.Insights/privateLinkScopes/deploy.bicep:1]: Private Link Scope -> private link scope

File: arm/Microsoft.Insights/components/deploy.bicep (hunk 1 above)
Comment (MrMCake) [reply arm/Microsoft.Insights/components/deploy.bicep]: removed the key output

File: arm/Microsoft.Insights/components/readme.md (hunk 2 above)
Comment (MrMCake) [reply arm/Microsoft.Insights/components/readme.md]: done

File: arm/Microsoft.Insights/metricAlerts/deploy.bicep (hunk 3 above)
Comment (MrMCake) [reply arm/Microsoft.Insights/metricAlerts/deploy.bicep:1]: done

File: arm/Microsoft.Insights/privateLinkScopes/.parameters/parameters.json (hunk 4 above)
Comment (MrMCake) [reply arm/Microsoft.Insights/privateLinkScopes/.parameters/parameters.json:5]: done

File: arm/Microsoft.Insights/privateLinkScopes/deploy.bicep (hunk 5 above)
Comment (MrMCake) [reply arm/Microsoft.Insights/privateLinkScopes/deploy.bicep:1]: done

**Review by eriqua — DISMISSED** [review]
🆗

File: arm/Microsoft.Insights/privateLinkScopes/deploy.bicep [hunk 6]
```diff
@@ -1,8 +1,8 @@
-@description('Required. Name of the Private Link Scope.')
+@description('Required. Name of the private link scope.')
 @minLength(1)
-param privateLinkScopeName string
+param name string
 
-@description('Optional. The location of the Private Link Scope. Should be global.')
+@description('Optional. The location of the private link ccope. Should be global.')
```
Comment (eriqua) [inline arm/Microsoft.Insights/privateLinkScopes/deploy.bicep:5]: typo scope

File: arm/Microsoft.Insights/privateLinkScopes/readme.md [hunk 7]
```diff
@@ -18,10 +18,10 @@ This module deploys Azure Monitor Private Link Scope
 | Parameter Name | Type | Default Value | Possible Values | Description |
 | :-- | :-- | :-- | :-- | :-- |
 | `cuaId` | string |  |  | Optional. Customer Usage Attribution id (GUID). This GUID must be previously registered |
-| `location` | string | `global` |  | Optional. The location of the Private Link Scope. Should be global. |
+| `location` | string | `global` |  | Optional. The location of the private link ccope. Should be global. |
```
Comment (eriqua) [inline arm/Microsoft.Insights/privateLinkScopes/readme.md:21]: typo scope

File: arm/Microsoft.Insights/activityLogAlerts/deploy.bicep [hunk 8]
```diff
@@ -1,5 +1,5 @@
 @description('Required. The name of the Alert.')
-param alertName string
+param name string
```
Comment (eriqua) [inline arm/Microsoft.Insights/activityLogAlerts/deploy.bicep:2]: Alert -> alert here as well. Didn't notice first, sorry

File: arm/Microsoft.Insights/privateLinkScopes/readme.md (hunk 7 above)
Comment (MrMCake) [reply arm/Microsoft.Insights/privateLinkScopes/readme.md]: done

File: arm/Microsoft.Insights/activityLogAlerts/deploy.bicep (hunk 8 above)
Comment (MrMCake) [reply arm/Microsoft.Insights/activityLogAlerts/deploy.bicep:2]: done

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**